pkg/controller/projectreference/projectreference_controller.go

```diff
@@ -121,67 +122,58 @@ func (r *ReconcileProjectReference) Reconcile(request reconcile.Request) (reconc
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileProjectReference) ReconcileHandler(adapter *ReferenceAdapter, reqLogger logr.Logger) (reconcile.Result, error) {
-	// Set conditions
-	prState, err := adapter.EnsureProjectReferenceInitialized()
-	if prState == ObjectModified || err != nil {
-		return r.requeueOnErr(err)
-	}
 
-	// Cleanup
-	if adapter.IsDeletionRequested() {
-		err := adapter.EnsureProjectCleanedUp()
-		if err != nil {
-			return r.requeueAfter(5*time.Second, err)
-		}
-		return r.doNotRequeue()
-	}
+type OperationResult struct {
+	RequeueDelay   time.Duration
+	RequeueRequest bool
+	CancelRequest  bool
+}
 
-	// Make projectReference  be processed based on state of ProjectClaim and Project Reference
-	claimStatus, err := adapter.EnsureProjectClaimReady()
-	if claimStatus == gcpv1alpha1.ClaimStatusReady || err != nil {
-		return r.requeueOnErr(err)
-	}
+func StopProcessing() (OperationResult, error) {
+	return OperationResult{0 * time.Second, false, true}, nil
```

Comment:
what about this?
```suggestion
	return OperationResult{0 * time.Second, RequeueRequest: false, CancelRequest : true}, nil
```

Reply:
As the struct is declared 10 lines above, I think this is not necessary as long as it stays with 3 parameters.

Reply:
I think that adding these inside the util function allows people to peek inside the fucntion and discern what boolean mean what without going to the  struct decleration

Reply:
Okay, I don't like the verboseness this creates, but I agree it might be easier to read. Will name the parameters.